doc/provisioning.puml

```diff
@@ -0,0 +1,31 @@
+@startuml
+(*) -->[user creates PVC] "PVC exists"
+-->[PV controller checks storage class and existing PVs, sets the\nvolume.beta.kubernetes.io/storage-provisioner annotation] "PVC ready for provisioning" as PVC_READY
```

Comment:
I think that PV controller does not add `volume.beta.kubernetes.io/storage-provisioner` right away. It's:

```
PVC exists:
  if  late binding:
      wait for pod + annotate the PVC
  else (immediate binding): 
      annotate the PVC right away
```

Reply:
You are right, I checked by watching PVC updates also for late binding. I've updated the diagram and also made it easier to read by styling the text.

When following the "unset selected node" arrow, it is obvious that the PV controller may see a PVC where it has already set the "provisioner" annotation earlier. That annotation is never reset, is it?

I changed the text to "if necessary, PV controller sets provisioner annotation" to reflect that it is possible and valid that the annotation already exists and then gets preserved.

I did that on *both* branches because expect that the code is the same.

Reply:
I moved the diagram out of this PR into PR #532